src/lib/fetch.ts

```diff
@@ -38,7 +38,7 @@ export function createFetch(got: Got): GotFetch {
     appendSearchParams(url.searchParams, defaultSearchParams);
 
     const mergedOptions = gotGlobal.mergeOptions(got.defaults.options, {
-      url,
+      url: url.toString(),
```

Comment:
I jsut ran the tests and one of them is failing
https://github.com/alexghr/got-fetch/blob/e7e7f0644120f45afce3ef779847efc5d5090d7a/src/test/fetch.request.test.ts#L51-L58

I've had issues in the past with this specific test where `got` sometimes messes up handling query string parameters. It seems to work if we tell `got` that there are not query string parameters for it to handle (which is kind of true because we merged them ourselves)

```suggestion
      url: url.toString(),
      searchParams: undefined,
```